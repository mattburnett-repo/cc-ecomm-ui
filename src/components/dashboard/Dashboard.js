import NavBar from '../nav/NavBar';
import BrowseSearch from '../nav/BrowseSearch';

export default function Dashboard () {
    const mockOrderData = [{
        "order": {
            "cart": {
                "cart_id": 1,
                "cart_items": [
                    {
                        "product_id": 1,
                        "product_name": "product_name_01",
                        "product_price": 1,
                        "product_quantity": 1,
                        "product_total_price": 1
                    },
                    {
                        "product_id": 2,
                        "product_name": "product_name_02",
                        "product_price": 2,
                        "product_quantity": 2,
                        "product_total_price": 4
                    },
                    {
                        "product_id": 3,
                        "product_name": "product_name_03",
                        "product_price": 3,
                        "product_quantity": 3,
                        "product_total_price": 9
                    },
                    {
                        "product_id": 4,
                        "product_name": "product_name_04",
                        "product_price": 4,
                        "product_quantity": 4,
                        "product_total_price": 16
                    }
                ]
            },
            "user_id": 1,
            "order_id": 1,
            "order_date": null,
            "total_price": {
                "sum": 30,
                "cart_id": 1
            }
        }
    }];

    const mockCartData = [
        {
            "id": 1,
            "product_id": 1,
            "name": "product_name_01",
            "product_quantity": 1,
            "product_price": "1.00",
            "line_item_total_price": "1.00"
        },
        {
            "id": 1,
            "product_id": 2,
            "name": "product_name_02",
            "product_quantity": 2,
            "product_price": "2.00",
            "line_item_total_price": "4.00"
        },
        {
            "id": 1,
            "product_id": 3,
            "name": "product_name_03",
            "product_quantity": 3,
            "product_price": "3.00",
            "line_item_total_price": "9.00"
        },
        {
            "id": 1,
            "product_id": 4,
            "name": "product_name_04",
            "product_quantity": 4,
            "product_price": "4.00",
            "line_item_total_price": "16.00"
        }
    ];

    return (
        <div>
            <NavBar calledFrom="dashboard" />
            <BrowseSearch />

            {/* FIXME: show actual data from mocks... */}
            <div role="presentation" aria-label="orders">
                {mockOrderData.map((item, index) => (
                    // <div role="presentation" aria-label="order">order_id: {item.order_id} user_id: {item.user_id} order_date: {item.order_date} total_price: {item.total_price.sum}</div>
                    <div role="presentation" aria-label='order' key={index}>order</div>
                ))}     
            </div>
            <div role="presentation" aria-label="carts">
                {mockCartData.map((item, index) => (
                    // <div role="presentation" aria-label="order">order_id: {item.order_id} user_id: {item.user_id} order_date: {item.order_date} total_price: {item.total_price.sum}</div>
                    <div role="presentation" aria-label='cart' key={index}>cart</div>
                ))}  
            </div>
        </div>
    )
}