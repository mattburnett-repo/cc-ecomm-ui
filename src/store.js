import { configureStore } from '@reduxjs/toolkit';

import authReducer from './features/auth/authSlice';
import productCategoryReducer from './features/productCategory/productCategorySlice'
import productsListingReducer from './features/products/productsSlice'
import ordersReducer from './features/orders/ordersSlice'
import cartsReducer from './features/carts/cartsSlice'

function rootReducer() {
    return {
        reducer: {
            auth: authReducer,
            productCategories: productCategoryReducer,
            orders: ordersReducer,
            carts: cartsReducer,
            products: productsListingReducer,
        }    
    }
}

const store = configureStore(rootReducer());

export default store;