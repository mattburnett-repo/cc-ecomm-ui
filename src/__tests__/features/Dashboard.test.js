import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';

import { Provider } from 'react-redux'
import store from '../../store'

import Dashboard from '../../features/dashboard/Dashboard';

describe('<Dashboard /> feature tests', () => {
    test.todo('it should have functional / integration tests')
    // beforeEach(() => render(<Provider store={store}><Dashboard /></Provider>));

    // it('displays a nav bar', () => {
    //     const theVal = screen.getAllByRole('presentation', {name: /nav-bar/i});
    //     expect(theVal).toHaveLength(1)
    // })
    // it('displays orders info', () => {
    //     const theVal = screen.getAllByRole('presentation', {name: /^orders$/i});
    //     expect(theVal.length).toBe(1);
    // })
    // it('displays carts info', () => {
    //     const theVal = screen.getAllByRole('presentation', {name: /^carts$/i});
    //     expect(theVal.length).toBe(1);
    // })
    
    // it('creates a snapshot', () => {
    //     const tree = renderer.create(<Provider store={store}><Dashboard /></Provider>).toJSON();
    //     expect(tree).toMatchSnapshot();  
    // })
}) // end feature

