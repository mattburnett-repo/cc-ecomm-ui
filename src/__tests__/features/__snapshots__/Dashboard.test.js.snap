// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> feature tests creates a snapshot 1`] = `
<div>
  <div
    aria-label="nav-bar"
    role="presentation"
  >
    <div
      aria-label="header-message"
      role="presentation"
    >
      dashboard
       
      Hello (username goes here)
    </div>
    <button
      aria-label="logout"
      onClick={[Function]}
    >
      Log Out
    </button>
  </div>
  <div
    aria-label="browse-search"
    role="presentation"
  >
    <select
      aria-label="product-categories"
      role="presentation"
    >
      <option
        value={4}
      >
        Auto
      </option>
      <option
        value={3}
      >
        Health
      </option>
      <option
        value={1}
      >
        Pets
      </option>
      <option
        value={2}
      >
        Tech
      </option>
    </select>
    <input
      aria-label="search-terms"
      placeholder="Enter search terms"
    />
    <button
      aria-label="search-button"
    >
      Search
    </button>
  </div>
  <div
    aria-label="orders"
    role="presentation"
  >
    <h2>
      OrdersDisplay
    </h2>
  </div>
  <div
    aria-label="carts"
    role="presentation"
  >
    <h2>
      Carts
    </h2>
    <div
      aria-label="cart"
      role="presentation"
    >
      cart_id: 
      1
       user_id: 
      1
    </div>
  </div>
</div>
`;
